subworkflows/local/metagenomics_postprocessing.nf

```diff
@@ -0,0 +1,118 @@
+include { MALTEXTRACT          } from '../../modules/nf-core/maltextract/main'
+include { AMPS                 } from '../../modules/nf-core/amps/main'
+include { TAXPASTA_MERGE       } from '../../modules/nf-core/taxpasta/merge/main'
+include { TAXPASTA_STANDARDISE } from '../../modules/nf-core/taxpasta/standardise/main'
+include { MEGAN_RMA2INFO       } from '../../modules/nf-core/megan/rma2info/main'
+
+workflow METAGENOMICS_POSTPROCESSING {
+
+    take:
+    ch_postprocessing_input // different between each profiling --> postprocessing tool,
+                            // defined in metagenomics profiling subworkflow
+
+    main:
+    ch_versions      = Channel.empty()
+    ch_multiqc_files = Channel.empty()
+
+    // For MALT we have an additional step that includes maltextract+amps
+    if ( params.metagenomics_run_postprocessing && params.metagenomics_profiling_tool == 'malt' ) {
+
+        tax_list = Channel.fromPath(params.metagenomics_maltextract_taxonlist)
+        ncbi_dir = Channel.fromPath(params.metagenomics_maltextract_ncbidir)
+
+        // Malt could have been executed multiple times (group_size paramter and strandedness)
+        // We want to combine the chunks, but run MaltExtract on double and singlestranded individually
+        ch_strandedness = ch_postprocessing_input
+            .transpose()
+            .map{ meta, reads ->
+                [
+                    meta+['id':"${meta.strandedness}stranded"],
+                    reads
+                ]
+            }
+            .groupTuple(by:0)
+
+        // could no be two entries in the channel, so combine with the tax_list and ncbi
+        ch_maltextract_input = ch_strandedness
+            .combine(tax_list)
+            .combine(ncbi_dir)
+            .multiMap{
+                rma6:[it[0],it[1]]
+                tax_list:it[2]
+                ncbi_dir:it[3]
+            }
+
+        //RUN MaltExtract
+        MALTEXTRACT ( ch_maltextract_input.rma6, ch_maltextract_input.tax_list, ch_maltextract_input.ncbi_dir)
+
+        // now we need to run AMPS for each MALTEXTRACT output
+        ch_amps_input = MALTEXTRACT.out.results.map{ it[1] }
+
+        AMPS ( ch_amps_input, ch_maltextract_input.tax_list, params.metagenomics_maltextract_filter )
```

Comment:
Does this work, when the first channel is not part of the `multiMap`? Getting mind bending here..

Reply:
Should work, the channel can be reused and has the same size as MALTEXTRACT.out.results content is always the same, so order doesnt matter

Reply:
OK... I'm not convinced but I'll trust you as I'm very tired so might be my brain isn't working